### ISSUE TITLE
No support for `;`, `and/&&`, `or/||` in subcommand `$()`

### ISSUE THREAD

**Issue by daniel-shimon** [xonsh/xonsh#4020]
Is this intentional?

## Expected Behavior

`echo $(echo a and echo b)` should print `a\nb`.

## Current Behavior

![image](https://user-images.githubusercontent.com/18242949/102327339-45917a80-3f8e-11eb-8a8a-4aefc48f4771.png)

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

**Comment by gforsyth** [comment]
Yeah, we don't support multiple commands inside a single capture block -- we can certainly put that on the list of things to do.

**Comment by anki-code** [comment]
Based on my current level of knowledge:

1. `(echo 1 ; echo 2)` interpreted by parser as `xonsh -c 'echo 1 ; echo 2'` so you can run `$((echo 1 ; echo 2))` (with `or`, `and`) or `$[(echo 1 ; echo 2)]` or `![(echo 1 ; echo 2)]`. But `!(())` is not working because of full capturing.

2. To implement this feature it's needed to significantly improve CommandPipeline because for now it supports `cmd | cmd | cmd` or `cmd &`. To implement sequence `<CommandPipeline> ; <CommandPipeline>` we need to improve xonsh/procs code.
